app/src/main/java/com/android/sample/ui/account/EditAccountScreen.kt

```diff
@@ -0,0 +1,451 @@
+        }
+  }
+}
+
+/**
+ * Method that creates the different fields of the user that can be read or modified
+ *
+ * @param newUserName the user name that will be displayed in the field as well as modified
+ * @param newDateOfBirth the date of birth that will be displayed in the field as well as modified
+ * @param modifier the modifier that will be applied on the column containing the different fields
+ * @param firebaseAuth the firebase authentication used to retrieve the users email address
+ * @param onValueChangeUserName the function which will be called when the user name is modified
+ * @param onValueChangeDateOfBirth the function which will be called when the date of birth is
+ *   modified
+ */
+@Composable
+private fun ListChangeableInformation(
+    newUserName: String,
+    newDateOfBirth: String?,
+    modifier: Modifier,
+    firebaseAuth: FirebaseAuth,
+    onValueChangeUserName: (String) -> Unit,
+    onValueChangeDateOfBirth: (String) -> Unit
+) {
+  Column(modifier = modifier, horizontalAlignment = Alignment.CenterHorizontally) {
+    InputTextBox(stringResource(R.string.username_label), newUserName, false, onValueChangeUserName)
+
+    Spacer(modifier = Modifier.weight(.3f))
+
+    InputTextBox(stringResource(R.string.email_label), firebaseAuth.currentUser!!.email!!, true) {}
+
+    Spacer(modifier = Modifier.weight(.3f))
+
+    DateOfBirthBox(newDateOfBirth, onValueChangeDateOfBirth)
+  }
+}
+
+/**
+ * Function that changes the new user name in the viewmodel as well as in the database
+ *
+ * @param userViewModel the view model of the user
+ * @param userName the old user name
+ * @param newUserName the new user name
+ */
+private fun saveChangesUsername(
+    userViewModel: UserViewModel,
+    userName: State<String?>,
+    newUserName: String
+) {
+  if (userName.value != newUserName) {
+    userViewModel.changeUserName(newUserName)
+  }
+}
+
+/**
+ * Function that changes the new profile picture in the viewmodel as well as in the database
+ *
+ * @param userViewModel the view model of the user
+ * @param imageRepositoryFirebase the firebase repository of the images
+ * @param userUid the uid of the user
+ * @param newProfilePictureImageBitmap the new profile picture of the user
+ */
+private fun saveChangesProfilePicture(
+    userViewModel: UserViewModel,
+    imageRepositoryFirebase: ImageRepositoryFirebase,
+    userUid: String,
+    newProfilePictureImageBitmap: ImageBitmap?
+) {
+  if (newProfilePictureImageBitmap != null) {
+    imageRepositoryFirebase.uploadImage(
+        userUid,
+        IMAGE_NAME,
+        ImageDirectoryType.USER,
+        newProfilePictureImageBitmap,
+        {
+          imageRepositoryFirebase.getImageUrl(
+              userUid,
+              IMAGE_NAME,
+              ImageDirectoryType.USER,
+              { userViewModel.changeProfilePictureUrl(it.toString()) },
+              { Log.e(LOG_MESSAGE_TAG, it.message!!) })
+        },
+        { Log.e(LOG_MESSAGE_TAG, it.message!!) })
```

Comment:
Wasn't there a way to do this in a single api call? I'm not sure tbh, so you can ignore this if there wasn't.

Reply:
Because we have to update the image stored for the profile picture and to update the profile picture of the user in his database with the new url to the storage where the image is stored, we need to have one call for updating the storage and one for retrieving the url.